### PR TITLE
Start adding a default system process

### PR DESCRIPTION
**Pull request by kasperl**
The system process will make it possible to have higher level
support for orchestrating container images and running containers
on the Toit platform.

With this, we can rewrite Jaguar as a pure user-level application
and we can start handling shared resources (network, etc.) better.

### CODE REVIEW

**Comment by kasperl** [comment]
So before this should land, we should agree on the structure and make sure we have the first test that runs with the new system process enabled.

File: system/boot.toit [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  container_manager := ContainerManager flash_registry rpc_broker
+  rpc_broker.install container_manager
+
+  // Set up RPC-based APIs.
+  ContainerApi rpc_broker container_manager
+
+  // TODO(kasper): Only start containers that require running on boot.
+  container_manager.images.do: | image/ContainerImage |
+    image.start
+  return container_manager
```
Comment (erikcorry) [inline system/boot.toit:47]: Add newline.

File: lib/system/containers.toit [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
```
Comment (erikcorry) [inline lib/system/containers.toit:4]:
```suggestion

// User-space side of the RPC API for installing container images in flash, and
// stopping and starting processes based on them.

```

File: system/api/containers.toit [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
```
Comment (erikcorry) [inline system/api/containers.toit:15]:
```suggestion

// Kernel-space side of the RPC API for installing container images in flash,
// and starting and stopping processes based on them.

```

File: system/boot.toit [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
```
Comment (erikcorry) [inline system/boot.toit:15]:
```suggestion

// Boot of the kernel.  Registers the RPC APIs and starts the default containers.

```

File: system/containers.toit [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
```
Comment (erikcorry) [inline system/containers.toit:15]:
```suggestion

// Implementation in the kernel of the API for installing container images in flash,
// and starting and stopping processes based on them.

```

**Review by erikcorry — APPROVED** [review]

File: system/boot.toit (hunk 1 above)
Comment (kasperl) [reply system/boot.toit]: Done.

File: lib/system/containers.toit (hunk 2 above)
Comment (kasperl) [reply lib/system/containers.toit:4]: Done.

File: system/containers.toit (hunk 5 above)
Comment (kasperl) [reply system/containers.toit:15]: Done.

File: system/boot.toit (hunk 4 above)
Comment (kasperl) [reply system/boot.toit:15]: Done.

File: system/boot.toit [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  flash_registry ::= FlashRegistry.scan
+  rpc_broker := SystemRpcBroker
+  container_manager := ContainerManager flash_registry rpc_broker
+  rpc_broker.install container_manager
+  ContainerApi rpc_broker container_manager  // Set up RPC-based APIs.
+  return container_manager
+
+/**
+Boot the system and run the necessary containers. Returns when the
+  containers have run to completetion or an error has occurred.
```
Comment (floitsch) [inline system/boot.toit:41]:
```suggestion
  containers have run to completion or an error has occurred.
```
```suggestion
  containers have run to completetion or an error has occurred.
```

File: system/containers.toit [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+    while container_is_running_ allocation_.offset allocation_.size:
+      result := container_kill_flash_image_ allocation_.offset allocation_.size
+      if result: attempts++
+      sleep --ms=10
+    manager.on_image_stop_all_ this
+
+  delete -> none:
+    stop_all
+    manager.image_registry.free allocation_
+    // TODO(kasper): Check in the other methods before using this?
```
Comment (floitsch) [inline system/containers.toit:117]: Still relevant?

Comment (floitsch) [inline system/containers.toit:117]: Still relevant?

**Review by floitsch — APPROVED** [review]
LGTM.

File: system/containers.toit (hunk 7 above)
Comment (kasperl) [reply system/containers.toit:117]: Yes, unfortunately.

**Pull request merged**